entity.go

```diff
@@ -177,6 +177,27 @@ func (e *EntityDefinition) EnsureValid() error {
 		keyNamesSeen[c.Name] = struct{}{}
 	}
 
+	if err := e.ensureNonNullablePrimaryKeys(); err != nil {
+		return errors.Wrap(err, "Nullable types are not allowed as primary key(s)")
```

Comment:
nit: not sure I'd wrap; you're not adding any additional info to the existing error. `primary key is of nullable type: "%q"` is enough of an error